integration_tests/web/test_web_client.py

```diff
@@ -0,0 +1,218 @@
+        self.assertIsNotNone(thread_reply)
+
+        modification = client.chat_update(channel=channel, ts=ts, text="Is this intentional?")
+        self.assertIsNotNone(modification)
+
+        reply_deletion = client.chat_delete(channel=channel, ts=thread_reply["ts"])
+        self.assertIsNotNone(reply_deletion)
+        message_deletion = client.chat_delete(channel=channel, ts=ts)
+        self.assertIsNotNone(message_deletion)
+
+    @async_test
+    async def test_basic_chat_operations_async(self):
+        client = self.async_client
+
+        auth = await client.auth_test()
+        self.assertIsNotNone(auth)
+        subdomain = auth["team"]
+
+        channel = self.channel_id
+        message = "This message was posted by <https://slack.dev/python-slackclient/|python-slackclient>! " + \
+                  "(integration_tests/test_web_client.py #test_chat_operations)"
+        new_message: SlackResponse = await client.chat_postMessage(channel=channel, text=message)
+        self.assertEqual(new_message["message"]["text"], message)
+        ts = new_message["ts"]
+
+        permalink = await client.chat_getPermalink(channel=channel, message_ts=ts)
+        self.assertIsNotNone(permalink)
+        self.assertRegex(permalink["permalink"], f"https://{subdomain}.slack.com/archives/{channel}/.+")
+
+        new_reaction = await client.reactions_add(channel=channel, timestamp=ts, name="eyes")
+        self.assertIsNotNone(new_reaction)
+
+        reactions = await client.reactions_get(channel=channel, timestamp=ts)
+        self.assertIsNotNone(reactions)
+
+        reaction_removal = await client.reactions_remove(channel=channel, timestamp=ts, name="eyes")
+        self.assertIsNotNone(reaction_removal)
+
+        thread_reply = await client.chat_postMessage(channel=channel, thread_ts=ts, text="threading...")
+        self.assertIsNotNone(thread_reply)
+
+        modification = await client.chat_update(channel=channel, ts=ts, text="Is this intentional?")
+        self.assertIsNotNone(modification)
+
+        reply_deletion = await client.chat_delete(channel=channel, ts=thread_reply["ts"])
+        self.assertIsNotNone(reply_deletion)
+        message_deletion = await client.chat_delete(channel=channel, ts=ts)
+        self.assertIsNotNone(message_deletion)
+
+    # -------------------------
+    # pagination
+
+    def test_pagination_with_iterator(self):
+        client = self.sync_client
+        fetched_count = 0
+        # SlackResponse is an iterator that fetches next if next_cursor is not ""
+        for response in client.conversations_list(limit=1, exclude_archived=1, types="public_channel"):
+            fetched_count += len(response["channels"])
+            if fetched_count > 1:
+                break
+
+        self.assertGreater(fetched_count, 1)
+
+    # TODO: Fix this issue
+    @pytest.mark.skip
+    @async_test
+    async def test_pagination_with_iterator_async(self):
```

Comment:
No one has reported this issue but `run_async=True` mode doesn't allow pagination using `next_cursor` iterators. I'm not sure about the best way to resolve this but at least, we need to change the internals not to call `run_until_complete`.